src/main/java/spoon/reflect/visitor/DefaultJavaPrettyPrinter.java

```diff
@@ -1773,4 +1795,35 @@ public void calculate(CompilationUnit sourceCompilationUnit, List<CtType<?>> typ
 	public Map<Integer, Integer> getLineNumberMapping() {
 		return printer.getLineNumberMapping();
 	}
+
+	/**
+	 * @return current implementation of {@link PrinterTokenWriter}
+	 */
+	public PrinterTokenWriter getPrinterTokenWriter() {
+		return printer.getTokenStream();
+	}
+	/**
+	 * Set {@link PrinterTokenWriter}, which has to be used to print tokens
+	 * The {@link PrinterTokenWriter#setNext(PrinterTokenWriter)} will be called
+	 * with internal implementation of {@link PrinterTokenWriter},
+	 * which delegates all tokens to {@link PrinterHelper}
+	 */
+	public DefaultJavaPrettyPrinter setPrinterTokenWriter(PrinterTokenWriter tokenWriter) {
```

Comment:
if the configuration happens in the helper, then do we need to have a setter there? this is confusing. The setter would only be in the delegate object (the helper probably)

Reply:
I want to make helper invisible/internal, so clients cannot call it. 
I expect that we will get rid of helper once in future, after DJPP code is refactored enough and produces nice print tokens directly.